### PR TITLE
vsphere: add storage secret field to cluster create request

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Add a separate secret id/name field to vSphere create request to allow using different url, username, pwd, datacenter, datastore, res. pool, folder settings for vSphere storage class setup. This is optional, by default all these setting come from the main cluster secret.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] OpenAPI and Postman files updated (if needed)

### CODE REVIEW

File: apis/pipeline/pipeline.yaml [hunk 1]
```diff
@@ -4447,9 +4447,17 @@ components:
...
                         - resourceGroup
                   properties:
+                        storageSecretId:
+                            type: string
+                            example: "62bc3c75-91fb-4670-bad4-24b401a9deac"
+                            description: "Secret ID used to setup VSphere storage classes. Overrides the default settings in main cluster secret."
+                        storageSecretName:
+                            type: string
+                            example: "my-vsphere-storage-secret"
+                            description: "Secret name used to setup VSphere storage classes. Overrides the default one in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4457]:
```suggestion
                            description: "Secret name used to setup VSphere storage classes. Overrides default value from the main cluster secret."
```

File: apis/pipeline/pipeline.yaml [hunk 2]
```diff
@@ -4447,9 +4447,17 @@ components:
                   required:
                         - resourceGroup
                   properties:
+                        storageSecretId:
+                            type: string
+                            example: "62bc3c75-91fb-4670-bad4-24b401a9deac"
+                            description: "Secret ID used to setup VSphere storage classes. Overrides the default settings in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4453]:
```suggestion
                            description: "Secret ID used to setup VSphere storage classes. Overrides default value from the main cluster secret."
```

File: apis/pipeline/pipeline.yaml [hunk 3]
```diff
@@ -4447,9 +4447,17 @@ components:
...
+                        storageSecretName:
+                            type: string
+                            example: "my-vsphere-storage-secret"
+                            description: "Secret name used to setup VSphere storage classes. Overrides the default one in main cluster secret."
                         folder:
                             type: string
-                            description: "Folder to create nodes in."
+                            description: "Folder to create nodes in. Overrides the default one in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4460]:
```suggestion
                            description: "Folder to create nodes in. Overrides default value from the main cluster secret."
```

File: apis/pipeline/pipeline.yaml [hunk 4]
```diff
@@ -4459,11 +4467,11 @@ components:
                                      summary: "Absolute path to the folder. Most vSphere installations use paths prefixed with /Datacenter/vm/."
                         datastore:
                             type: string
-                            description: "Name of datastore or datastore cluster to place VM disks on."
+                            description: "Name of datastore or datastore cluster to place VM disks on. Overrides the default one in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4470]:
```suggestion
                            description: "Name of datastore or datastore cluster to place VM disks on. Overrides default value from the main cluster secret."
```

File: apis/pipeline/pipeline.yaml [hunk 5]
```diff
@@ -4459,11 +4467,11 @@ components:
...
                             example: Datastore-cluster
                         resourcePool:
                             type: string
-                            description: "Virtual machines will be created in this resource pool."
+                            description: "Virtual machines will be created in this resource pool. Overrides the default one in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4474]:
```suggestion
                            description: "Virtual machines will be created in this resource pool. Overrides default value from the main cluster secret."
```

File: apis/pipeline/pipeline.yaml [hunk 6]
```diff
@@ -4505,7 +4513,7 @@ components:
                     description: MiBs of RAM to attach to each node.
                 template:
                     type: string
-                    description: "Name of VM template available on vSphere to clone as the base of nodes."
+                    description: "Name of VM template available on vSphere to clone as the base of nodes. Overrides the default one in main cluster secret."
```
Comment (siliconbrain) [inline apis/pipeline/pipeline.yaml:4516]:
```suggestion
                    description: "Name of VM template available on vSphere to clone as the base of nodes. Overrides default value from the main cluster secret."
```

File: internal/providers/vsphere/pke/driver/cluster_creator.go [hunk 7]
```diff
@@ -24,6 +24,8 @@ import (
 	"go.uber.org/cadence/client"
 	corev1 "k8s.io/api/core/v1"
 
+	"github.com/banzaicloud/pipeline/internal/secret/secrettype"
```
Comment (siliconbrain) [inline internal/providers/vsphere/pke/driver/cluster_creator.go:27]: Organize imports.

File: internal/providers/vsphere/pke/driver/cluster_creator.go [hunk 8]
```diff
@@ -333,6 +351,19 @@ func (p VspherePKEClusterCreationParamsPreparer) Prepare(ctx context.Context, pa
 	return nil
 }
 
+func (p VspherePKEClusterCreationParamsPreparer) verifySecretIsOfType(orgID uint, secretID string, secretType string) error {
+	if secretID != "" {
```
Comment (siliconbrain) [inline internal/providers/vsphere/pke/driver/cluster_creator.go:355]: Prefer early returns:
```go
if secretID == "" {
	return nil
}
```

File: internal/providers/vsphere/pke/workflow/create_node_activity.go [hunk 9]
```diff
@@ -25,9 +25,13 @@ import (
 	"github.com/ghodss/yaml"
 	"github.com/vmware/govmomi/find"
 	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
 	"github.com/vmware/govmomi/vim25/types"
 	"go.uber.org/cadence/activity"
 
+	"github.com/banzaicloud/pipeline/internal/providers/pke/pkeworkflow"
```
Comment (siliconbrain) [inline internal/providers/vsphere/pke/workflow/create_node_activity.go:32]: Organize imports.

File: internal/providers/vsphere/pke/driver/cluster_creator.go [hunk 10]
```diff
@@ -24,6 +24,8 @@ import (
 	"go.uber.org/cadence/client"
 	corev1 "k8s.io/api/core/v1"
 
+	"github.com/banzaicloud/pipeline/internal/secret/secrettype"
+
```
Comment (sagikazarmark) [inline internal/providers/vsphere/pke/driver/cluster_creator.go:28]:
```suggestion
```

**Pull request merged**